src/renderer/components/subscriptions-live/subscriptions-live.js

```diff
@@ -0,0 +1,515 @@
+      }))).flatMap((o) => o)
+      videoList.push(...videoListFromRemote)
+
+      this.updateVideoListAfterProcessing(videoList)
+      this.isLoading = false
+      this.updateShowProgressBar(false)
+    },
+
+    updateVideoListAfterProcessing(videoList) {
+      // Filtering and sorting based in preference
+      videoList.sort((a, b) => {
+        if (b.liveNow) {
+          return 1
+        }
+
+        if (a.liveNow) {
+          return -1
+        }
+
+        if (b.premiereDate && a.premiereDate) {
+          return b.premiereDate - a.premiereDate
+        }
+
+        if (b.premiereDate && !a.premiereDate) {
+          return 1
+        }
+
+        if (!b.premiereDate && a.premiereDate) {
+          return -1
+        }
+
+        if (typeof a.publishedDate === 'undefined' || b.publishedDate === 'undefined') {
+          return calculatePublishedDate(b.publishedText) - calculatePublishedDate(a.publishedText)
+        } else {
+          return b.publishedDate - a.publishedDate
+        }
+      })
+      if (this.hideLiveStreams) {
+        videoList = videoList.filter(item => {
+          return (!item.liveNow && !item.isUpcoming)
+        })
+      }
+      if (this.hideUpcomingPremieres) {
+        videoList = videoList.filter(item => {
+          if (item.isRSS) {
+            // viewCount is our only method of detecting premieres in RSS
+            // data without sending an additional request.
+            // If we ever get a better flag, use it here instead.
+            return item.viewCount !== '0'
+          }
+          // Observed for premieres in Local API Subscriptions.
+          return item.premiereDate == null
+        })
+      }
+
+      this.videoList = videoList.filter((video) => {
+        if (this.hideWatchedSubs) {
+          const historyIndex = this.historyCache.findIndex((x) => {
+            return x.videoId === video.videoId
+          })
+
+          return historyIndex === -1
+        } else {
+          return true
+        }
+      })
+    },
+
+    maybeLoadVideosForSubscriptionsFromRemote: async function () {
+      if (this.fetchSubscriptionsAutomatically) {
+        // `this.isLoading = false` is called inside `loadVideosForSubscriptionsFromRemote` when needed
+        await this.loadVideosForSubscriptionsFromRemote()
+      } else {
+        this.videoList = []
+        this.attemptedFetch = false
+        this.isLoading = false
+      }
+    },
+
+    getChannelLiveLocal: async function (channel, failedAttempts = 0) {
+      try {
+        const channelInstance = await getLocalChannel(channel.id)
+
+        if (channelInstance.has_live_streams) {
+          const liveTab = await channelInstance.getLiveStreams()
+
+          if (liveTab.videos === null) {
+            this.errorChannels.push(channel)
+            return []
+          }
+
+          return parseLocalChannelVideos(liveTab.videos, channelInstance.header.author)
+        } else {
+          return []
+        }
```

Comment:
Reduce no. of requests by half like video tab (2 => 1 request per channel)
```suggestion
        const entries = await getLocalChannelLiveStreams(channel.id)

        if (entries === null) {
          this.errorChannels.push(channel)
          return []
        }

        return entries
```

Plus adding following code to `src/renderer/helpers/api/local.js`
```js
export async function getLocalChannelLiveStreams(id) {
  const innertube = await createInnertube()

  try {
    const response = await innertube.actions.execute(Endpoints.BrowseEndpoint.PATH, Endpoints.BrowseEndpoint.build({
      browse_id: id,
      params: 'EgdzdHJlYW1z8gYECgJ6AA%3D%3D'
      // protobuf for the live tab (this is the one that YouTube uses,
      // it has some empty fields in the protobuf but it doesn't work if you remove them)
    }))

    const liveStreamsTab = new YT.Channel(null, response)

    // if the channel doesn't have a live tab, YouTube returns the home tab instead
    // so we need to check that we got the right tab
    if (liveStreamsTab.current_tab?.endpoint.metadata.url?.endsWith('/streams')) {
      return parseLocalChannelVideos(liveStreamsTab.videos, liveStreamsTab.header.author)
    } else {
      return []
    }
  } catch (error) {
    console.error(error)
    if (error instanceof Utils.ChannelError) {
      return null
    } else {
      throw error
    }
  }
}
```